cli/src/cmd/mod.rs

```diff
@@ -191,41 +190,28 @@ fn get_artifact_from_name(
 }
 
 /// Find using src/ContractSource.sol:ContractName
-// TODO: Is there a better / more ergonomic way to get the artifacts given a project and a
-// path?
 fn get_artifact_from_path(
     project: &Project,
-    path: String,
-    name: String,
+    contract_path: String,
+    contract_name: String,
 ) -> eyre::Result<(Abi, CompactBytecode, CompactDeployedBytecode)> {
     // Get sources from the requested location
-    let abs_path = dunce::canonicalize(PathBuf::from(path))?;
-    let mut sources = Sources::new();
-    sources.insert(abs_path.clone(), Source::read(&abs_path)?);
-
-    // Get artifact from the contract name and sources
-    let mut config = SolFilesCache::builder().insert_files(sources.clone(), None)?;
-    config.files.entry(abs_path).and_modify(|f| f.artifacts = vec![name.clone()]);
-
-    let artifacts = config
-        .read_artifacts::<MinimalCombinedArtifacts>(project.artifacts_path())?
-        .into_values()
-        .collect::<Vec<_>>();
-
-    if artifacts.is_empty() {
-        eyre::bail!("could not find artifact")
-    } else if artifacts.len() > 1 {
-        eyre::bail!("duplicate contract name in the same source file")
-    }
-    let artifact = artifacts[0].clone();
+    let abs_path = dunce::canonicalize(PathBuf::from(contract_path))?;
+
+    let cache = SolFilesCache::read_joined(&project.paths)?;
```

Comment:
So does the cache end up being the main entrpoint to reading an artifact?

Reply:
yeah, because the cache now contains the actual path to the artifacts of a solidity file,
so we have `file` -> (`contract` -> [(`solc version`, `path`)])

```json
      "artifacts": {
        "Dapp": {
          "0.8.11+commit.d7f03943.Darwin.appleclang": "Dapp.sol/Dapp.json"
        }
      }
```

Reply:
Got it - in general I'd expect the cache to not be a user facing component? Maybe we can just document it as intended usage of the lib to avoid confusion? Like "Common ethers solc workflows" (not a blocker for this PR, just saying)

Reply:
yeah, sg, should be added to the docs and examples in ethers-solc